tools/storage/src/read_snapshot.rs

```diff
@@ -101,8 +135,131 @@ async fn test_deserialize_account() {
     this_path.push("fixtures/state_epoch_79_ver_33217173.795d/state.manifest");
     let snapshot_manifest = load_snapshot_manifest(&this_path).expect("parse manifest");
     let archive_path = this_path.parent().unwrap();
-    let r = accounts_from_snapshot_backup(snapshot_manifest, &archive_path)
+    let account_states = accounts_from_snapshot_backup(snapshot_manifest, &archive_path)
         .await
         .expect("could not parse snapshot");
-    dbg!(&r);
+    for account_state in account_states.iter() {
+        println!("account_address: {:?}", account_state.get_account_address());
+        let r = account_state.get_coin_store_resource();
+        println!("coin store: {:?}", r);
+    }
+}
+
+use libra_types::legacy_types::legacy_recovery::{AccountRole};
+use diem_types::validator_config::ValidatorConfig;
+
+use libra_types::legacy_types::{
+    ancestry_legacy::LegacyAncestryResource,
+    cumulative_deposits::{CumulativeDepositResource, LegacyBalanceResource},
+    legacy_address::LegacyAddress,
+    legacy_currency_info::CurrencyInfoResource,
+    legacy_miner_state::TowerStateResource,
+    makewhole_resource::MakeWholeResource,
+    receipts::ReceiptsResource,
+    validator_config::ValidatorConfigResource,
+    wallet::{CommunityWalletsResourceLegacy, SlowWalletListResource, SlowWalletResource},
+};
+
+use std::convert::TryInto;
+use diem_types::account_config::AccountResource;
+use libra_types::legacy_types::validator_config::ConfigResource;
+
+#[derive(Debug, Default, Clone)]
+pub struct LegacyRecoveryV6 {
+    ///
+    pub account: Option<AccountAddress>,
+    ///
+    pub auth_key: Option<AuthenticationKey>,
+    ///
+    pub role: AccountRole,
+    ///
+    pub balance: Option<LegacyBalanceResource>,
+    ///
+    pub val_cfg: Option<ValidatorConfigResource>,
+    ///
+    pub miner_state: Option<TowerStateResource>,
+    ///
+    pub comm_wallet: Option<CommunityWalletsResourceLegacy>,
+
+    pub currency_info: Option<CurrencyInfoResource>,
+    ///
+    pub ancestry: Option<LegacyAncestryResource>,
+    ///
+    pub make_whole: Option<MakeWholeResource>,
+    ///
+    pub receipts: Option<ReceiptsResource>,
+    ///
+    pub cumulative_deposits: Option<CumulativeDepositResource>,
+    ///
+    pub slow_wallet: Option<SlowWalletResource>,
+    ///
+    pub slow_wallet_list: Option<SlowWalletListResource>,
+    // TODO: use on V7 tools
+    // ///
+    // pub fullnode_counter: Option<FullnodeCounterResource>,
+    // ///
+    // pub autopay: Option<AutoPayResource>,
+}
+
+
+fn get_legacy_recovery(account_states: &Vec<AccountState>) -> anyhow::Result<Vec<LegacyRecoveryV6>> {
+    let mut legacy_recovery_v6 = Vec::new();
+    for account_state in account_states.iter() {
```

Comment:
Please add for USERS
Vouches
Donor_Voice_Structs
FeeMaker
Jail

for 0x1 Framework State:
ValidatorUniverse
SlowWalletRegistry
DonorVoiceRegistry
Account::OriginatingAddress (Migrate...) these are Tables... not sure how to do.

Reply:
so far added support for 
```
Burn::UserBurnPreference
vouch::*
donor_voice_txs::*
fee_maker::FeeMaker
jail::Jail
pledge_account::*
```